drivers/dma/dma_mcux_edma.c

```diff
@@ -24,17 +24,37 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/irq.h>
 
+#ifdef CONFIG_DMA_MCUX_EDMA_LEGACY
 #define DT_DRV_COMPAT nxp_mcux_edma
+#elif CONFIG_DMA_MCUX_EDMA_V3
+#define DT_DRV_COMPAT nxp_mcux_edma_v3
+#endif
 
 LOG_MODULE_REGISTER(dma_mcux_edma, CONFIG_DMA_LOG_LEVEL);
 
+#define HAS_CHANNEL_GAP(inst) DT_INST_NODE_HAS_PROP(inst, channel_gap) ||
+
+#define DMA_MCUX_HAS_CHANNEL_GAP (DT_INST_FOREACH_STATUS_OKAY(HAS_CHANNEL_GAP) 0)
+
 struct dma_mcux_edma_config {
 	DMA_Type *base;
-	DMAMUX_Type *dmamux_base;
+	DMAMUX_Type **dmamux_base;
+#ifdef DMA_MCUX_HAS_CHANNEL_GAP
+	uint32_t channel_gap[2];
+#endif
+	uint8_t channels_per_mux;
 	int dma_channels; /* number of channels */
 	void (*irq_config_func)(const struct device *dev);
 };
 
+#if defined FSL_FEATURE_EDMA_HAS_ERROR_IRQ && FSL_FEATURE_EDMA_HAS_ERROR_IRQ
```

Comment:
can this put into the soc PLATFORM_HAS_ERROR_IRQ? put feature dependences in driver  is not that good.

Reply:
But how about the offset between channels share the same interrupt, we need to use dt? I also see this kind of macros (`FSL_FEATURE`) are using in various mcux shim drivers

Reply:
use dts is a good selection, the legacy code with FSL_FEATURE is histric burden.

Reply:
Is it okay if we treat this as an improvement as the Code Freeze is coming? the dt can be used, but i need to add new properties, then update the driver and dt for all mcux SoC that are affected

Reply:
@hakehuang is this ok for you?

Reply:
@Dat-NguyenDuy . Hake is suggesting replacing the `FSL_FEATURE` with something you define in `soc.h`. His suggestion is to use `PLATFORM_HAS_ERROR_IRQ`. I feel that should be similar and an easy change.

Reply:
I've updated, not just that minor. @hakehuang can you pls revisit

Reply:
we want you to put things in soc level, not in driver config, and what's worse is that you define a marco that skip our code, why you wlways change others code clogic and not consider to make your code fit in existing code?

Reply:
Re-pushed as discussed in this https://github.com/zephyrproject-rtos/zephyr/pull/61311#discussion_r1333835313